### PR TITLE
Change groupId to a fixed value. Change artifactId to provide default values. #215

### PR DESCRIPTION
**Pull request by btyanagawamg**
Please review #215 .

Since groupId can not be set freely, it changed to a fixed value.
Although artifactId can be freely set, it changed to prepare default values for usability improvement.

### CODE REVIEW

File: secure-login-demo/scripts/create-app.sh [hunk 1]
```diff
@@ -3,13 +3,24 @@
...
-#   ${ARTIFACT_ID}=Artifact ID of tutorial project. This project's ARTIFACT_ID must be set 'secure-login'.
 #   ${VERSION}=Version of tutorial project.
 #   ${HOST_IP}=IP address for access to this web application. Defaults to 'localhost'.
 #   ${APSRV_WEB_PORT}=Port number for access to this web application. Defaults to '8080'.
 #   ${APSRV_H2DB_PORT}=Port number for access to H2DB. Defaults to '9212'.
 
+case "$ARCHETYPE_ARTIFACT_ID" in
+    terasoluna-gfw-multi-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.securelogin
+        export ARTIFACT_ID=secure-login
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/create-app.sh]: - [ ] You should not use fixed artifact id. If test will be failed, you must correct test cases.

File: secure-login-demo/scripts/create-app.sh [hunk 2]
```diff
@@ -3,13 +3,24 @@
 # Required variables:
 #   ${ARCHETYPE_ARTIFACT_ID}=Artifact ID of blank project's archetype.
 #   ${ARCHETYPE_VERSION}=Version of blank project's archetype.
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/create-app.sh]: - [ ] It seems it's better to use default value (tutorial-apps's artifact version) when not specified.

File: secure-login-demo/scripts/create-app.sh [hunk 3]
```diff
@@ -3,13 +3,24 @@
...
-#   ${GROUP_ID}=Group ID of tutorial project. This project's GROUP_ID must be set 'com.example.securelogin'.
-#   ${ARTIFACT_ID}=Artifact ID of tutorial project. This project's ARTIFACT_ID must be set 'secure-login'.
 #   ${VERSION}=Version of tutorial project.
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/create-app.sh]: - [ ] It seems it's better to use default value (specified version at guideline) when not specified.

File: first-spring-security/scripts/create-app.sh [hunk 4]
```diff
@@ -3,10 +3,26 @@
...
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3}
+        ;;
+    terasoluna-gfw-multi-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3-multi}
+        ;;
+    * )
+        echo "You can not select the specified ARCHETYPE_ARTIFACT_ID."
+        exit 1
```
Comment (yoshikawaa) [inline first-spring-security/scripts/create-app.sh]: - [ ] Please test `exit 1` is affective at executing travis-ci script.

File: first-spring-security/scripts/create-app.sh [hunk 5]
```diff
@@ -3,10 +3,26 @@
...
+    terasoluna-gfw-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3}
+        ;;
+    terasoluna-gfw-multi-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3-multi}
+        ;;
+    * )
+        echo "You can not select the specified ARCHETYPE_ARTIFACT_ID."
```
Comment (yoshikawaa) [inline first-spring-security/scripts/create-app.sh]: - [ ] same mean [select * specify], and I think you should show speficied value.

File: secure-login-demo/scripts/create-app.sh (hunk 1 above)
Comment (btyanagawamg) [reply secure-login-demo/scripts/create-app.sh]: af4973459a88da5a4e2993b82c0ad9fe91ed902a
`ARTIFACT_ID` has been modified to be freely selectable in above commit.

File: secure-login-demo/scripts/create-app.sh (hunk 2 above)
Comment (btyanagawamg) [reply secure-login-demo/scripts/create-app.sh]: f4e9326e5cddb9db5a73ef770afd585506666bb7 
I prepared a default artifact version in above commit.

File: secure-login-demo/scripts/create-app.sh (hunk 3 above)
Comment (btyanagawamg) [reply secure-login-demo/scripts/create-app.sh]: f4e9326e5cddb9db5a73ef770afd585506666bb7 
I prepared a default tutorial application version in above commit.

File: first-spring-security/scripts/create-app.sh (hunk 5 above)
Comment (btyanagawamg) [reply first-spring-security/scripts/create-app.sh]: edf9f7621f48c05b5070d1da7cdabc3446febb97 
With the above commit, I fixed the log message.

File: .travis.yml [hunk 6]
```diff
@@ -10,30 +10,29 @@ cache:
 env:
   global:
     - ARCHETYPE_VERSION=5.5.0-SNAPSHOT
```
Comment (yoshikawaa) [inline .travis.yml]: - [ ] Not required. If not specify ARCHETYPE_VERSION, will be used default version.

File: secure-login-demo/scripts/convert-secure-login-css.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (yoshikawaa) [inline secure-login-demo/scripts/convert-secure-login-css.sh]: - [ ] Unrelated changes to this issue.

File: first-spring-security/scripts/create-app.sh [hunk 8]
```diff
@@ -2,11 +2,30 @@
...
+    terasoluna-gfw-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3}
+        ;;
+    terasoluna-gfw-multi-web-blank-mybatis3-archetype )
+        export GROUP_ID=com.example.security
+        export ARTIFACT_ID=${ARTIFACT_ID:=first-spring-security-mybatis3-multi}
+        ;;
+    * )
+        echo "[ERROR] You can not set ${ARCHETYPE_ARTIFACT_ID} for ARCHETYPE_ARTIFACT_ID."
```
Comment (yoshikawaa) [inline first-spring-security/scripts/create-app.sh]: - [ ] It seems to be better as follows.

`ARCHETYPE_ARTIFACT_ID [${ARCHETYPE_ARTIFACT_ID}] is not available in this tutorial.`

File: .travis.yml (hunk 6 above)
Comment (btyanagawamg) [reply .travis.yml]: `ARCHETYPE_VERSION` is needed by [`change-maven-settings.sh`](https://github.com/terasolunaorg/tutorial-apps/blob/master/common/scripts/change-maven-settings.sh), and `change-maven-settings.sh` does not called from create-app.sh .
So `ARCHETYPE_VERSION` is need .

File: first-spring-security/scripts/create-app.sh (hunk 4 above)
Comment (btyanagawamg) [reply first-spring-security/scripts/create-app.sh]: I checked at https://github.com/terasolunaorg/tutorial-apps/commit/b5b1edcac2ee126fb4ce78ecb27bc52ddbaf3f8e commit, and got result below .
https://api.travis-ci.org/v3/job/346595984/log.txt
Certainly, travis did not stop even if `exit 1` came back.

File: secure-login-demo/scripts/convert-secure-login-css.sh (hunk 7 above)
Comment (btyanagawamg) [reply secure-login-demo/scripts/convert-secure-login-css.sh]: I will correspond above comment in #225 ISSUE .

File: first-spring-security/scripts/create-app.sh [hunk 9]
```diff
@@ -2,11 +2,29 @@
...
 # Required variables:
 #   ${ARCHETYPE_ARTIFACT_ID}=Artifact ID of blank project's archetype.
-#   ${ARCHETYPE_VERSION}=Version of blank project's archetype.
-#   ${GROUP_ID}=Group ID of tutorial project. This project's GROUP_ID must be set 'com.example.security'.
-#   ${ARTIFACT_ID}=Artifact ID of tutorial project.
-#   ${VERSION}=Version of tutorial project.
-
+#   ${ARCHETYPE_VERSION}=Version of blank project's archetype. If not specified, the default value is set.
+#   ${ARTIFACT_ID}=Artifact ID of tutorial project. If not specified, the default value is set.
+#   ${VERSION}=Version of tutorial project. If not specified, the default value is set.
```
Comment (yoshikawaa) [inline first-spring-security/scripts/create-app.sh]: - [ ] Please fix docs/README.md.
https://github.com/terasolunaorg/tutorial-apps/tree/d2710738830a98868031ad4c835095e85d46f4f7/docs#how-to-execute

File: first-spring-security/scripts/create-app.sh (hunk 8 above)
Comment (btyanagawamg) [reply first-spring-security/scripts/create-app.sh]: a629786d3b128329bf45b10b353d2a448698b1e1 
Fixed at above commit .

File: first-spring-security/scripts/create-app.sh (hunk 9 above)
Comment (btyanagawamg) [reply first-spring-security/scripts/create-app.sh]: 82eb130175d845bf49c4cba64a70202853ca7963 
Fixed at above commit .

File: docs/README.md [hunk 10]
```diff
@@ -36,15 +36,14 @@ First of all, you must set following environment variables.
...
-| ARCHETYPE_VERSION | Version of blank project's archetype. | (e.g. ``5.3.0.RELEASE``) |
-| GROUP_ID | Group ID of tutorial project. | You must set to same value as tutorial instructions explained in Development Guideline. |
-| ARTIFACT_ID | Artifact ID of tutorial project. | Basically, set to same value as tutorial instructions explained in Development Guideline. |
-| VERSION | Version of tutorial project. | Basically, set to same value as tutorial instructions explained in Development Guideline. |
 
 You can also set following optional environment variables.
 
 | Environment Variable | Description | Note |
 |:------------- |:----------------- |:----------------- |
+| ARCHETYPE_VERSION | Version of blank project's archetype. | Defaults to tutorial-apps pom.xml verion.|
```
Comment (yoshikawaa) [inline docs/README.md]: - [ ] as follows.

e.g. ``5.3.0.RELEASE``
Defaults to specified version  in tutorial-apps pom.xml.

File: docs/README.md [hunk 11]
```diff
@@ -36,15 +36,14 @@ First of all, you must set following environment variables.
...
+| ARTIFACT_ID | Artifact ID of tutorial project. | If not specified, the default value(depending on each archetype and type of application) is set. |
```
Comment (yoshikawaa) [inline docs/README.md]: - [ ] as follws.

Defaults to value as tutorial instructions explained in Development Guideline. (e.g. `todo`)

File: docs/README.md [hunk 12]
```diff
@@ -36,15 +36,14 @@ First of all, you must set following environment variables.
...
 |:------------- |:----------------- |:----------------- |
 | ARCHETYPE_ARTIFACT_ID | Artifact ID of blank project's archetype. | You can choose archetypes from tutorial instructions explained in Development Guideline. |
-| ARCHETYPE_VERSION | Version of blank project's archetype. | (e.g. ``5.3.0.RELEASE``) |
-| GROUP_ID | Group ID of tutorial project. | You must set to same value as tutorial instructions explained in Development Guideline. |
-| ARTIFACT_ID | Artifact ID of tutorial project. | Basically, set to same value as tutorial instructions explained in Development Guideline. |
-| VERSION | Version of tutorial project. | Basically, set to same value as tutorial instructions explained in Development Guideline. |
 
 You can also set following optional environment variables.
 
 | Environment Variable | Description | Note |
```
Comment (yoshikawaa) [inline docs/README.md]: - [ ] Please re-construct table with column as required/optional.

File: docs/README.md (hunk 10 above)
Comment (btyanagawamg) [reply docs/README.md]: b6c22846f4af722baa323e1da8f23b8975ab4382 
Fixed at above commit .

File: docs/README.md (hunk 11 above)
Comment (btyanagawamg) [reply docs/README.md]: b6c22846f4af722baa323e1da8f23b8975ab4382 
Fixed at above commit .

File: docs/README.md (hunk 12 above)
Comment (btyanagawamg) [reply docs/README.md]: b6c22846f4af722baa323e1da8f23b8975ab4382 
Fixed at above commit .

File: docs/README.md (hunk 10 above)
Comment (yoshikawaa) [reply docs/README.md]: - [ ] `5.4.0.RELEASE` is wrong version. ( not released... )

Comment (btyanagawamg) [reply docs/README.md]: 5805c0a45d56a41f40bc2c9987f48cf7924d65a1
Revert version .

File: first-spring-security/scripts/create-app.sh (hunk 4 above)
Comment (yoshikawaa) [reply first-spring-security/scripts/create-app.sh]: @btyanagawamg 
Other repositories do not matter.
It is pointed out that consistency is not established among the shells created here.
If you respond with another issue, let's create an Issue and let it be done.

Comment (btyanagawamg) [reply first-spring-security/scripts/create-app.sh]: @yoshikawaa 

I think it best to add `|| exit $?` after `bash ./${TARGET_TUTORIAL}/scripts/create-app.sh` command .
`exit 1` will stop travis-ci script soon .
Please see 51f705f11662b4719500419d1efc27ef57380035 change and result .

Comment (yoshikawaa) [reply first-spring-security/scripts/create-app.sh]: Regardless of other repository stories,
Is it about whether to respect the return code of create-app.sh, should we respect the identity with the behavior at Jenkins, or should we respect the default behavior of Travis CI's Script?
Whichever we adopted, there was no big advantage or disadvantage, and it was concluded that we should respect the default of Travis CI.
